internal/reconciliations/oathkeeper/deployment.go

```diff
@@ -0,0 +1,105 @@
+package oathkeeper
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"github.com/avast/retry-go/v4"
+	"github.com/kyma-project/api-gateway/apis/operator/v1alpha1"
+	"github.com/kyma-project/api-gateway/internal/clusterconfig"
+	"github.com/kyma-project/api-gateway/internal/reconciliations"
+	"github.com/kyma-project/api-gateway/internal/reconciliations/oathkeeper/maester"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+const (
+	deploymentName = "ory-oathkeeper"
+)
+
+//go:embed deployment_light.yaml
+var deploymentLight []byte
+
+//go:embed deployment.yaml
+var deployment []byte
+
+func reconcileOathkeeperDeployment(ctx context.Context, k8sClient client.Client, apiGatewayCR v1alpha1.APIGateway) error {
+
+	clusterSize, err := clusterconfig.EvaluateClusterSize(ctx, k8sClient)
+	if err != nil {
+		return err
+	}
+
+	ctrl.Log.Info("Reconciling Ory Oathkeeper Deployment", "Cluster size", clusterSize, "name", deploymentName, "Namespace", reconciliations.Namespace)
+
+	if apiGatewayCR.IsInDeletion() {
+		return deleteDeployment(k8sClient, deploymentName)
+	}
+
+	if clusterSize == clusterconfig.Evaluation {
+		return reconcileDeployment(ctx, k8sClient, deploymentName, &deploymentLight)
+	}
+	return reconcileDeployment(ctx, k8sClient, deploymentName, &deployment)
+}
+
+func reconcileDeployment(ctx context.Context, k8sClient client.Client, name string, deploymentManifest *[]byte) error {
+
+	ctrl.Log.Info("Reconciling Deployment", "name", name, "Namespace", reconciliations.Namespace)
+	templateValues := make(map[string]string)
+	templateValues["Name"] = name
+	templateValues["Namespace"] = reconciliations.Namespace
+	templateValues["ServiceAccountName"] = maester.ServiceAccountName
+
+	err := reconciliations.ApplyResource(ctx, k8sClient, *deploymentManifest, templateValues)
+	if err != nil {
+		return err
+	}
+
+	return retry.Do(func() error {
+		var podList corev1.PodList
+		err := k8sClient.List(context.Background(), &podList, client.MatchingLabels{
+			"app.kubernetes.io/instance": "ory",
+			"app.kubernetes.io/name":     "oathkeeper",
+		})
+
+		if err != nil {
+			return err
+		}
+
+		for _, pod := range podList.Items {
+			if pod.Status.Phase != "Running" {
```

Comment:
I think we should be able to use a constant here.
```suggestion
			if pod.Status.Phase != corev1.PodRunning {
```